src/Model/AuditReaderInterface.php

```diff
@@ -19,46 +19,28 @@
 interface AuditReaderInterface
 {
     /**
-     * @param string $className
-     * @param string $id
-     * @param string $revision
-     *
-     * @return object
+     * @param mixed $id
      */
-    public function find($className, $id, $revision);
+    public function find(string $className, $id, int $revision): ?object;
 
     /**
-     * @param string $className
-     * @param int    $limit
-     * @param int    $offset
-     *
      * @return object[]
      */
-    public function findRevisionHistory($className, $limit = 20, $offset = 0);
+    public function findRevisionHistory(string $className, int $limit = 20, int $offset = 0): array;
 
-    /**
-     * @param string $classname
-     * @param string $revision
-     *
-     * @return object
-     */
-    public function findRevision($classname, $revision);
+    public function findRevision(string $className, int $revision): object;
```

Comment:
I cant see the implementation but are we sure there is always a Revision ?
?object as return type seems more logic to me

Reply:
To do that, which seems more logic to me too, we will need to catch some exceptions on sonata doctrine orm, because we using Entity audit bundle, and it throws exceptions instead (and we should not couple here with those exceptions)

Reply:
This could be deprecated soon https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1068#discussion_r453310312

But I look at the code, and I don't think catching exception will be an issue.
It will be better since the current interface hasn't @throws tag.

We should
- Change `find` return typehint to ?object
- Change `findRevsion` return typehint to ?Revision
- Add a NEXT_MAJOR issue/comment to doctrine orm

Do you agree ?

Reply:
Agree on the find part, Not sure about the second one. If I add ?Revisión I had to add dependency here with entityAudit, and not sure if that is a great idea. Other persistence bundles could implement this in a different way. Agree on the third point, we need an issue to track this idea of deprecating the entityAudit and revisions part (not sure if we should deprecate them, but we can discuss it there).

Reply:
Oh, ok. findRevision should return `?object`.

Reply:
I created https://github.com/sonata-project/SonataDoctrineORMAdminBundle/issues/1085

Reply:
Fixed the typehints